### PR TITLE
Feature/nox 171 source filtering

### PR DESCRIPTION
**Pull request by jan-schutte**

### CODE REVIEW

File: .idea/.idea.Nox.Generator.dir/.idea/.gitignore
Comment (rochar) [inline .idea/.idea.Nox.Generator.dir/.idea/.gitignore:1]: do you want to commite .idea configuration, maybe add this to the main gitignore only

File: src/Nox.Generator/Infrastructure/Persistence/DtoDbContext.template.cs [hunk 1]
```diff
@@ -10,8 +10,10 @@
 using Nox.Infrastructure;
 using Nox.Infrastructure.Persistence;
 
+{{- if codeGeneratorState.Solution.Domain != null }}
```
Comment (rochar) [inline src/Nox.Generator/Infrastructure/Persistence/DtoDbContext.template.cs:13]: The Generator will not call template if the domain is null

File: src/Nox.Integration.SqlServer/Extensions/SqlServerIntegrationOptionsExtension.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using Nox.Types.EntityFramework.Abstractions;
+using Nox.Types.EntityFramework.Enums;
+
+namespace Nox.Integration.SqlServer;
+
+public static class SqlServerIntegrationOptionsExtension
+{
+    public static NoxIntegrationOptionsBuilder WithSqlServerStore(this NoxIntegrationOptionsBuilder optionsBuilder)
+    {
+        optionsBuilder.Services.AddSingleton<INoxDatabaseProvider>(provider => new SqlServerDatabaseProvider(
```
Comment (rochar) [inline src/Nox.Integration.SqlServer/Extensions/SqlServerIntegrationOptionsExtension.cs:16]: like we agree lets have a `IEntityDatabaseProvider: INoxDatabaseProvider`

and `IIntegrationDatabaseProvider: INoxdatabaseProvider` for example

**Review by rochar — CHANGES_REQUESTED** [review]

**Comment by jan-schutte** [comment]
redundant

**Pull request closed without merging**